contracts/triggers/AaveTrailingQuotePriceTrigger.sol

```diff
@@ -0,0 +1,245 @@
+                (, encompassingFeedTimestamp1) = getRoundInfo(
+                    triggerSubData.quoteTokenAddr,
+                    encompassingFeedRoundId1
+                );
+                encompassedFeedTimestamp = baseMaxPriceTimeStamp;
+            }
+
+            if (!roundEncompassed(
+                encompassingFeedRoundId0,
+                encompassingFeedRoundId1,
+                encompassingFeedTimestamp0,
+                encompassingFeedTimestamp1,
+                encompassedFeedTimestamp
+            )) return false;
+        }
+
+        address[] memory assets = new address[](2);
+        assets[0] = triggerSubData.baseTokenAddr;
+        assets[1] = triggerSubData.quoteTokenAddr;
+        uint256[] memory currAssetPrices = aaveOracleV3.getAssetsPrices(
+            assets
+        );
+
+        uint256 baseCurrPrice = currAssetPrices[0];
+        uint256 quoteCurrPrice = currAssetPrices[1];
+
+        uint256 currPrice = baseCurrPrice * 1e8 / quoteCurrPrice;
+        uint256 maxPrice = baseMaxPrice * 1e8 / quoteMaxPrice;
+
+        return checkPercentageDiff(currPrice, maxPrice, triggerSubData.percentage);
+    }
+
+    /// @dev checking if both maxRoundIds were 'latest' at the same time
+    function roundEncompassed(
+        uint80 encompassingFeedRoundId0,
+        uint80 encompassingFeedRoundId1,
+        uint256 encompassingFeedTimestamp0,
+        uint256 encompassingFeedTimestamp1,
+        uint256 encompassedFeedTimestamp
+    ) internal pure returns (bool) {
+        // encompassed feed round has to be in between the encompassing feed rounds
+        if (encompassingFeedTimestamp0 > encompassedFeedTimestamp) return false;
+
+        uint256 nextRoundId = encompassingFeedRoundId0 + 1;
+        uint16 phaseId = uint16(encompassingFeedRoundId0 >> 64);
+        uint256 nextPhaseRoundId = ((phaseId + 1) << 64) + 1;
+
+        // encompassingFeedRoundId1 not valid next roundId
+        if (encompassingFeedRoundId1 != nextRoundId
+            && encompassingFeedRoundId1 != nextPhaseRoundId) return false;
+
+        // encompassed feed round has to be in between the encompassing feed rounds
+        if (encompassingFeedTimestamp1 < encompassedFeedTimestamp) return false;
+
+        return true;
+    }
+
+    /// @notice Given the currentPrice and the maxPrice see if there diff. > than percentage
+    function checkPercentageDiff(
+        uint256 _currPrice,
+        uint256 _maxPrice,
+        uint256 _percentage
+    ) public pure returns (bool) {
+        uint256 amountDiff = (_maxPrice * _percentage) / 10**10;
+
+        return _currPrice <= (_maxPrice - amountDiff);
+    }
+
+    /// @dev Helper function that returns chainlink price data
+    /// @param _inputTokenAddr Token address we are looking the usd price for
+    /// @param _roundId Chainlink roundId, if 0 uses the latest
+    function getRoundInfo(address _inputTokenAddr, uint80 _roundId)
```

Comment:
Can use TokenPriceHelper.getRoundInfo() if we change the function to inject ` IAggregatorV3(aggregatorAddr)` as first param.

Reply:
good idea, it is done